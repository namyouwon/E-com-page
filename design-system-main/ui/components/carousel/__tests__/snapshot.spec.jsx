/* eslint-env jest */
import React from 'react';
import Carousel from '../';
import createHelpers from '../../../../jest.helpers';

const { matchesMarkup } = createHelpers(__dirname);

it('renders a carousel, first tab active', () =>
  matchesMarkup(<Carousel panelActive="1" />));

it('renders a carousel, second tab active', () =>
  matchesMarkup(<Carousel panelActive="2" />));

it('renders a carousel, third tab active', () =>
  matchesMarkup(<Carousel panelActive="3" />));

it('renders a carousel, auto-play enabled', () =>
  matchesMarkup(<Carousel panelActive="1" autoPlay />));

it('renders a carousel, auto-play disabled', () =>
  matchesMarkup(<Carousel panelActive="1" autoPlay="stop" />));
