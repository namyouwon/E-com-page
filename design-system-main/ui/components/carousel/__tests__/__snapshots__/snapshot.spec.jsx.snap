// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a carousel, auto-play disabled 1`] = `
<div
  className="slds-carousel"
>
  <div
    className="slds-carousel__stage"
  >
    <span
      className="slds-carousel__autoplay"
    >
      <button
        aria-pressed={true}
        className="slds-button slds-button_icon slds-button_icon-border-filled slds-button_icon-x-small"
        onClick={[Function]}
        title="Stop auto-play"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#right"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Stop auto-play
        </span>
      </button>
    </span>
    <div
      className="slds-carousel__panels"
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        aria-hidden="false"
        aria-labelledby="indicator-id-28"
        className="slds-carousel__panel"
        id="content-id-25"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="0"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Visit App Exchange"
              src="/assets/images/carousel/carousel-01.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Visit App Exchange
            </h2>
            <p>
              Extend Salesforce with the #1 business marketplace.
            </p>
          </div>
        </a>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="indicator-id-29"
        className="slds-carousel__panel"
        id="content-id-26"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="-1"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Click to Customize"
              src="/assets/images/carousel/carousel-02.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Click to Customize
            </h2>
            <p>
              Use the Object Manager to add fields, build layouts, and more.
            </p>
          </div>
        </a>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="indicator-id-30"
        className="slds-carousel__panel"
        id="content-id-27"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="-1"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Download SalesforceA"
              src="/assets/images/carousel/carousel-03.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Download SalesforceA
            </h2>
            <p>
              Get the mobile app that's just for Salesforce admins.
            </p>
          </div>
        </a>
      </div>
    </div>
    <ul
      className="slds-carousel__indicators"
      role="tablist"
    >
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-25"
          aria-selected="true"
          className="slds-carousel__indicator-action slds-is-active"
          href="#"
          id="indicator-id-28"
          onClick={[Function]}
          role="tab"
          tabIndex="0"
          title="Visit App Exchange tab"
        >
          <span
            className="slds-assistive-text"
          >
            Visit App Exchange tab
          </span>
        </a>
      </li>
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-26"
          aria-selected="false"
          className="slds-carousel__indicator-action"
          href="#"
          id="indicator-id-29"
          onClick={[Function]}
          role="tab"
          tabIndex="-1"
          title="Click to Customize tab"
        >
          <span
            className="slds-assistive-text"
          >
            Click to Customize tab
          </span>
        </a>
      </li>
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-27"
          aria-selected="false"
          className="slds-carousel__indicator-action"
          href="#"
          id="indicator-id-30"
          onClick={[Function]}
          role="tab"
          tabIndex="-1"
          title="Download SalesforceA tab"
        >
          <span
            className="slds-assistive-text"
          >
            Download SalesforceA tab
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders a carousel, auto-play enabled 1`] = `
<div
  className="slds-carousel"
>
  <div
    className="slds-carousel__stage"
  >
    <span
      className="slds-carousel__autoplay"
    >
      <button
        aria-pressed={false}
        className="slds-button slds-button_icon slds-button_icon-border-filled slds-button_icon-x-small"
        onClick={[Function]}
        title="Stop auto-play"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#pause"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Stop auto-play
        </span>
      </button>
    </span>
    <div
      className="slds-carousel__panels"
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        aria-hidden="false"
        aria-labelledby="indicator-id-22"
        className="slds-carousel__panel"
        id="content-id-19"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="0"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Visit App Exchange"
              src="/assets/images/carousel/carousel-01.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Visit App Exchange
            </h2>
            <p>
              Extend Salesforce with the #1 business marketplace.
            </p>
          </div>
        </a>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="indicator-id-23"
        className="slds-carousel__panel"
        id="content-id-20"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="-1"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Click to Customize"
              src="/assets/images/carousel/carousel-02.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Click to Customize
            </h2>
            <p>
              Use the Object Manager to add fields, build layouts, and more.
            </p>
          </div>
        </a>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="indicator-id-24"
        className="slds-carousel__panel"
        id="content-id-21"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="-1"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Download SalesforceA"
              src="/assets/images/carousel/carousel-03.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Download SalesforceA
            </h2>
            <p>
              Get the mobile app that's just for Salesforce admins.
            </p>
          </div>
        </a>
      </div>
    </div>
    <ul
      className="slds-carousel__indicators"
      role="tablist"
    >
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-19"
          aria-selected="true"
          className="slds-carousel__indicator-action slds-is-active"
          href="#"
          id="indicator-id-22"
          onClick={[Function]}
          role="tab"
          tabIndex="0"
          title="Visit App Exchange tab"
        >
          <span
            className="slds-assistive-text"
          >
            Visit App Exchange tab
          </span>
        </a>
      </li>
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-20"
          aria-selected="false"
          className="slds-carousel__indicator-action"
          href="#"
          id="indicator-id-23"
          onClick={[Function]}
          role="tab"
          tabIndex="-1"
          title="Click to Customize tab"
        >
          <span
            className="slds-assistive-text"
          >
            Click to Customize tab
          </span>
        </a>
      </li>
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-21"
          aria-selected="false"
          className="slds-carousel__indicator-action"
          href="#"
          id="indicator-id-24"
          onClick={[Function]}
          role="tab"
          tabIndex="-1"
          title="Download SalesforceA tab"
        >
          <span
            className="slds-assistive-text"
          >
            Download SalesforceA tab
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders a carousel, first tab active 1`] = `
<div
  className="slds-carousel"
>
  <div
    className="slds-carousel__stage"
  >
    <div
      className="slds-carousel__panels"
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        aria-hidden="false"
        aria-labelledby="indicator-id-4"
        className="slds-carousel__panel"
        id="content-id-1"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="0"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Visit App Exchange"
              src="/assets/images/carousel/carousel-01.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Visit App Exchange
            </h2>
            <p>
              Extend Salesforce with the #1 business marketplace.
            </p>
          </div>
        </a>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="indicator-id-5"
        className="slds-carousel__panel"
        id="content-id-2"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="-1"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Click to Customize"
              src="/assets/images/carousel/carousel-02.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Click to Customize
            </h2>
            <p>
              Use the Object Manager to add fields, build layouts, and more.
            </p>
          </div>
        </a>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="indicator-id-6"
        className="slds-carousel__panel"
        id="content-id-3"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="-1"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Download SalesforceA"
              src="/assets/images/carousel/carousel-03.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Download SalesforceA
            </h2>
            <p>
              Get the mobile app that's just for Salesforce admins.
            </p>
          </div>
        </a>
      </div>
    </div>
    <ul
      className="slds-carousel__indicators"
      role="tablist"
    >
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-1"
          aria-selected="true"
          className="slds-carousel__indicator-action slds-is-active"
          href="#"
          id="indicator-id-4"
          onClick={[Function]}
          role="tab"
          tabIndex="0"
          title="Visit App Exchange tab"
        >
          <span
            className="slds-assistive-text"
          >
            Visit App Exchange tab
          </span>
        </a>
      </li>
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-2"
          aria-selected="false"
          className="slds-carousel__indicator-action"
          href="#"
          id="indicator-id-5"
          onClick={[Function]}
          role="tab"
          tabIndex="-1"
          title="Click to Customize tab"
        >
          <span
            className="slds-assistive-text"
          >
            Click to Customize tab
          </span>
        </a>
      </li>
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-3"
          aria-selected="false"
          className="slds-carousel__indicator-action"
          href="#"
          id="indicator-id-6"
          onClick={[Function]}
          role="tab"
          tabIndex="-1"
          title="Download SalesforceA tab"
        >
          <span
            className="slds-assistive-text"
          >
            Download SalesforceA tab
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders a carousel, second tab active 1`] = `
<div
  className="slds-carousel"
>
  <div
    className="slds-carousel__stage"
  >
    <div
      className="slds-carousel__panels"
      style={
        Object {
          "transform": "translateX(-100%)",
        }
      }
    >
      <div
        aria-hidden="true"
        aria-labelledby="indicator-id-10"
        className="slds-carousel__panel"
        id="content-id-7"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="-1"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Visit App Exchange"
              src="/assets/images/carousel/carousel-01.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Visit App Exchange
            </h2>
            <p>
              Extend Salesforce with the #1 business marketplace.
            </p>
          </div>
        </a>
      </div>
      <div
        aria-hidden="false"
        aria-labelledby="indicator-id-11"
        className="slds-carousel__panel"
        id="content-id-8"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="0"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Click to Customize"
              src="/assets/images/carousel/carousel-02.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Click to Customize
            </h2>
            <p>
              Use the Object Manager to add fields, build layouts, and more.
            </p>
          </div>
        </a>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="indicator-id-12"
        className="slds-carousel__panel"
        id="content-id-9"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="-1"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Download SalesforceA"
              src="/assets/images/carousel/carousel-03.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Download SalesforceA
            </h2>
            <p>
              Get the mobile app that's just for Salesforce admins.
            </p>
          </div>
        </a>
      </div>
    </div>
    <ul
      className="slds-carousel__indicators"
      role="tablist"
    >
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-7"
          aria-selected="false"
          className="slds-carousel__indicator-action"
          href="#"
          id="indicator-id-10"
          onClick={[Function]}
          role="tab"
          tabIndex="-1"
          title="Visit App Exchange tab"
        >
          <span
            className="slds-assistive-text"
          >
            Visit App Exchange tab
          </span>
        </a>
      </li>
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-8"
          aria-selected="true"
          className="slds-carousel__indicator-action slds-is-active"
          href="#"
          id="indicator-id-11"
          onClick={[Function]}
          role="tab"
          tabIndex="0"
          title="Click to Customize tab"
        >
          <span
            className="slds-assistive-text"
          >
            Click to Customize tab
          </span>
        </a>
      </li>
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-9"
          aria-selected="false"
          className="slds-carousel__indicator-action"
          href="#"
          id="indicator-id-12"
          onClick={[Function]}
          role="tab"
          tabIndex="-1"
          title="Download SalesforceA tab"
        >
          <span
            className="slds-assistive-text"
          >
            Download SalesforceA tab
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`renders a carousel, third tab active 1`] = `
<div
  className="slds-carousel"
>
  <div
    className="slds-carousel__stage"
  >
    <div
      className="slds-carousel__panels"
      style={
        Object {
          "transform": "translateX(-200%)",
        }
      }
    >
      <div
        aria-hidden="true"
        aria-labelledby="indicator-id-16"
        className="slds-carousel__panel"
        id="content-id-13"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="-1"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Visit App Exchange"
              src="/assets/images/carousel/carousel-01.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Visit App Exchange
            </h2>
            <p>
              Extend Salesforce with the #1 business marketplace.
            </p>
          </div>
        </a>
      </div>
      <div
        aria-hidden="true"
        aria-labelledby="indicator-id-17"
        className="slds-carousel__panel"
        id="content-id-14"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="-1"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Click to Customize"
              src="/assets/images/carousel/carousel-02.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Click to Customize
            </h2>
            <p>
              Use the Object Manager to add fields, build layouts, and more.
            </p>
          </div>
        </a>
      </div>
      <div
        aria-hidden="false"
        aria-labelledby="indicator-id-18"
        className="slds-carousel__panel"
        id="content-id-15"
        role="tabpanel"
      >
        <a
          className="slds-carousel__panel-action slds-text-link_reset"
          href="#"
          onClick={[Function]}
          tabIndex="0"
        >
          <div
            className="slds-carousel__image"
          >
            <img
              alt="Download SalesforceA"
              src="/assets/images/carousel/carousel-03.jpg"
            />
          </div>
          <div
            className="slds-carousel__content"
          >
            <h2
              className="slds-carousel__content-title"
            >
              Download SalesforceA
            </h2>
            <p>
              Get the mobile app that's just for Salesforce admins.
            </p>
          </div>
        </a>
      </div>
    </div>
    <ul
      className="slds-carousel__indicators"
      role="tablist"
    >
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-13"
          aria-selected="false"
          className="slds-carousel__indicator-action"
          href="#"
          id="indicator-id-16"
          onClick={[Function]}
          role="tab"
          tabIndex="-1"
          title="Visit App Exchange tab"
        >
          <span
            className="slds-assistive-text"
          >
            Visit App Exchange tab
          </span>
        </a>
      </li>
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-14"
          aria-selected="false"
          className="slds-carousel__indicator-action"
          href="#"
          id="indicator-id-17"
          onClick={[Function]}
          role="tab"
          tabIndex="-1"
          title="Click to Customize tab"
        >
          <span
            className="slds-assistive-text"
          >
            Click to Customize tab
          </span>
        </a>
      </li>
      <li
        className="slds-carousel__indicator"
        role="presentation"
      >
        <a
          aria-controls="content-id-15"
          aria-selected="true"
          className="slds-carousel__indicator-action slds-is-active"
          href="#"
          id="indicator-id-18"
          onClick={[Function]}
          role="tab"
          tabIndex="0"
          title="Download SalesforceA tab"
        >
          <span
            className="slds-assistive-text"
          >
            Download SalesforceA tab
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
